gasometer/src/lib.rs

```diff
@@ -249,13 +253,18 @@ impl<'config> Gasometer<'config> {
 				access_list_storage_len,
 				authorization_list_len,
 			} => {
+				self.inner_mut()?
+					.metrics
+					.init(zero_data_len, non_zero_data_len, false);
+
 				#[deny(clippy::let_and_return)]
 				let cost = self.config.gas_transaction_call
-					+ zero_data_len as u64 * self.config.gas_transaction_zero_data
-					+ non_zero_data_len as u64 * self.config.gas_transaction_non_zero_data
-					+ access_list_address_len as u64 * self.config.gas_access_list_address
-					+ access_list_storage_len as u64 * self.config.gas_access_list_storage_key
-					+ authorization_list_len as u64 * self.config.gas_per_empty_account_cost;
+					+ calldata_cost(zero_data_len, non_zero_data_len, self.config)
+					+ access_list_cost(
+						access_list_address_len,
+						access_list_storage_len,
+						self.config,
+					) + authorization_list_cost(authorization_list_len, self.config);
```

Comment:
Lets avoid refactoring code in PR's that implement new features. Maybe create an issue specific to refactoring this in a follow-up PR.